mlflow/models/evaluation/base.py

```diff
@@ -229,14 +257,7 @@ def data(self):
         if self._data is not None:
             return self._data
 
-        try:
-            from pyspark.sql import DataFrame as SparkDataFrame
-
-            spark_df_type = SparkDataFrame
-        except ImportError:
-            spark_df_type = None
-
-        if spark_df_type and isinstance(self._original_data, spark_df_type):
+        if self._spark_df_type and isinstance(self._original_data, self._spark_df_type):
             self._data = self._original_data.limit(
                 EvaluationDataset.SPARK_DATAFRAME_LIMIT
             ).toPandas()
```

Comment:
Have we verified how this behaves if the feature to evaluate is actually contained in the output from a VectorAssembler transformer stage? If the feature set is a single-column SparseVector or DenseVector, translating back to columns with a provided set of field names might be a useful quality of life feature. 
If this is completely out of scope, disregard this comment.

Reply:
All kinds of columns includes "pyspark.ml.linalg.Vector" type are supported.